### DISCUSSION TITLE
Which SVD algorithms do we use

### DISCUSSION THREAD

**Discussion by thijssteel** [tlapack/tlapack#216] [General]
For the SVD, which algorithms should we implement?

LAPACK has:
- implicit QR
- dqds
- bisection

There is also MRRR but this is not implemented in LAPACK (I think). I don't know the method, but people seem to refer to it as fast.

I'm wondering, do we need to support each of these or are some of them outdated?

From what I've found, the QR algorithm is the most accurate so we should probably have at least that. The other ones are faster, maybe we just use one of those?

GESVD also seems to be mostly level 2, should we try to use level 3 blas for the singular vectors (this seems relatively easy to do by accumulating the Givens rotations)?

**Comment by thijssteel** [comment]
After talking with some colleagues, we definitely need QR as some of their research applications depend on the increased accuracy.

**Comment by langou** [comment]
Good. Sorry for not following up earlier. It feels like Bidiagonal SVD QR is the way to go as a first step. I agree. Jim Demmel also agrees. We spoke about this a few days ago. Sorry for not following up earlier.
